include/lattice_field.h

```diff
@@ -72,18 +78,24 @@ namespace quda {
     QudaGhostExchange ghostExchange;
 
     /** The extended field radius (if applicable) */
-    int r[QUDA_MAX_DIM];
+    lat_dim_t r;
 
     /** For fixed-point fields that need a global scaling factor */
     double scale;
 
     /**
        @brief Default constructor for LatticeFieldParam
     */
-    LatticeFieldParam()
-    : precision(QUDA_INVALID_PRECISION), ghost_precision(QUDA_INVALID_PRECISION), nDim(4), pad(0),
-      siteSubset(QUDA_INVALID_SITE_SUBSET), mem_type(QUDA_MEMORY_DEVICE),
-      ghostExchange(QUDA_GHOST_EXCHANGE_PAD), scale(1.0)
+    LatticeFieldParam() :
```

Comment:
Wondering if now is a good time to adpot https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rc-default

Reply:
Agree, will apply

Reply:
8724b25a41a95a9d5b6066379e4fff0beab7efdd